### PR TITLE
chore: fix yeoman generator

### PR DESCRIPTION
**Pull request by juliodialpad**
# Fix Yeoman generator

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will not increment the version number, but will still deploy to documentation site on release:

- [ ] Documentation
- [ ] Build system
- [ ] CI
- [ ] Style (code style changes, not css changes)
- [ ] Test
- [x] Other (chore)

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1481

## :book: Description

- Moved Yeoman generator to the top of the workspace so we can create Generators for other stuff, not only for dialtone-vue
- Updated test.ejs with the new tests template created by @iropolo 
- Updated documentation, TL;DR: you can run the generator in two ways:
  - Inside packages/dialtone-vueX directory: `pnpm yo @dialpad/dialtone:vueX`
  - From the workspace root: `pnpm --filter ./packages/dialtone-vueX exec yo @dialpad/dialtone:vueX`
- Separated Yeoman generator in two folders vue2 and vue3 this gives us the possibility to use different templates from both versions
- Linked index.js inside generators/vue2 to the vue3 version so we only need to update one and use different templates.

## :bulb: Context

- After the monorepo migration Yeoman stopped working.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [ ] I have added / updated unit tests.
- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script.
- [ ] I have validated components with a screen reader.
- [ ] I have validated components keyboard navigation.

## :crystal_ball: Next Steps

- Maybe we can migrate Yeoman to [NX generators](https://nx.dev/features/generate-code)

### CODE REVIEW

File: generator-dialtone/generators/vue3/templates/story.ejs [hunk 1]
```diff
@@ -5,8 +5,8 @@
     as the storybook control defined in the corresponding `.story.js` file.
   -->
   <<%= componentNameKebab %>
-    :generic-prop="$attrs.genericProp"
-    @generic-event="$attrs.onGenericEvent"
+    :generic-prop="genericProp"
+    @generic-event="onGenericEvent"
```
Comment (braddialpad) [inline generator-dialtone/generators/vue3/templates/story.ejs]: We need to still include attrs don't we? or did you make it so this is handled elsewhere?

**Review by braddialpad — COMMENTED** [review]
Nice! this is probably the final thing we need to do to make contributing as straightforward as it was before the monorepo migration.

File: generator-dialtone/generators/vue3/templates/story.ejs (hunk 1 above)
Comment (juliodialpad) [reply generator-dialtone/generators/vue3/templates/story.ejs]: Right, forgot to update templates. 

Fixed on latest commit.

**Review by braddialpad — CHANGES_REQUESTED** [review]
I tested generation and found a couple syntax problems in the test.js

this test context thing:
Screenshot 2024-02-26 at 4 09 22 PM

something wrong with unclosed braces here:
Screenshot 2024-02-26 at 4 10 39 PM

**Comment by juliodialpad** [comment]
> I tested generation and found a couple syntax problems in the test.js
> 
> this test context thing: Screenshot 2024-02-26 at 4 09 22 PM
> 
> something wrong with unclosed braces here: Screenshot 2024-02-26 at 4 10 39 PM

Thanks for catching that! fixed a bunch of issues found, please review again.

File: generator-dialtone/generators/vue2/templates/story.ejs [hunk 2]
```diff
@@ -17,14 +17,17 @@
...
       same name.
     -->
-    <template v-if="defaultSlot">
-      <span v-html="defaultSlot" />
+    <template>
+      <span
+        v-if="$attrs.default"
+        v-html="$attrs.default"
+      />
     </template>
```
Comment (braddialpad) [inline generator-dialtone/generators/vue2/templates/story.ejs]: We can remove this `<template>` tag now that we no longer need a v-if

**Review by braddialpad — APPROVED** [review]

File: generator-dialtone/generators/vue2/templates/story.ejs (hunk 2 above)
Comment (juliodialpad) [reply generator-dialtone/generators/vue2/templates/story.ejs]: Right, forgot about that, done

**Pull request merged**